app/services/bgs/dependent_service.rb

```diff
@@ -78,5 +97,26 @@ def get_form_hash_686c(file_number:)
         }
       }
     end
+
+    def validate_file_number_format!(file_number:)
+      # We've observed cases where BGS's file numbers are ten or thirteen digits.
+      # BGS has indicated that these file numbers are incorrect and that file numbers
+      # should be eight or nine digits. I want to log cases like this, so we can
+      # tell BGS the kinds of file numbers we are seeing.
+      if file_number.length < 8 || file_number.length > 9
+        file_number_pattern = file_number.gsub(/[0-9]/, 'X')
+        raise "Aborting Form 686c submission: BGS file_nbr has invalid format! (#{file_number_pattern})"
```

Comment:
I think we are discouraged from raising PII in error messages. Like, file number could be an SSN, so we don't want to raise this error with an SSN in it and have the error and SSN logged. 

for example, if the SSN is `123-456-7889`, and `file_number == 'ssn:123-456-7889'` or  `file_number == '  
               123-456-7889'`, it would get logged here which I do not think is allowed.

I know this makes troubleshooting difficult, but they do not want PII in the logs if possible, or so I have heard. Just something to consider.